manala.locales/lookup_plugins/manala_locales_codes.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.plugins.lookup import LookupBase
+from ansible.errors import AnsibleError
+
+# As seen on native locale_gen module
+LOCALE_NORMALIZATION = {
+    ".utf8": ".UTF-8",
+    ".eucjp": ".EUC-JP",
+    ".iso885915": ".ISO-8859-15",
+    ".cp1251": ".CP1251",
+    ".koi8r": ".KOI8-R",
+    ".armscii8": ".ARMSCII-8",
+    ".euckr": ".EUC-KR",
+    ".gbk": ".GBK",
+    ".gb18030": ".GB18030",
+    ".euctw": ".EUC-TW",
+}
+
+class LookupModule(LookupBase):
+
+    # As seen on native locale_gen module
+    def normalize(self, name):
+        """locale -a might return the encoding in either lower or upper case.
+        Passing through this function makes them uniform for comparisons."""
+        for s, r in LOCALE_NORMALIZATION.items():
+            name = name.replace(s, r)
+        return name
+
+    def denormalize(self, name):
+        """locale -a might return the encoding in either lower or upper case.
+        Passing through this function makes them uniform for comparisons."""
+        for s, r in LOCALE_NORMALIZATION.items():
+            name = name.replace(r, s)
+        return name
+
+    def run(self, terms, variables=None, **kwargs):
+
+        results = []
+
+        wantcodedenormalized = kwargs.pop('wantcodedenormalized', False)
+        wantstatecurrent     = kwargs.pop('wantstatecurrent', False)
+
+        codes         = self._flatten(terms[0])
+        codesPresents = self._flatten(terms[1])
+
+        itemDefault = {
+            'state': 'present'
+        }
+
+        for code in codes:
+
+            items = []
+
+            # Short syntax
+            if isinstance(code, basestring):
+                item = itemDefault.copy()
+                item.update({
+                    'code': code
+                })
+            else:
+
+                # Must be a dict
+                if not isinstance(code, dict):
+                    raise AnsibleError('Expect a dict')
```

Comment:
s/Expect/Expected everywhere (assertion occurred already)

Reply:
See: https://github.com/manala/ansible-roles/issues/97 (and thx for the review)